From 4343148aafd040a3f39e092959faed4ac995f267 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:43 -0400
Subject: [infiniband] ocrdma: Allow DPP QP creation

Message-id: <fb6eda6ed76b35bd14c85a837a6d10764d68b9db.1411571481.git.dledford@redhat.com>
Patchwork-id: 96651
O-Subject: [Patch RHEL7 08/48] RDMA/ocrdma: Allow DPP QP creation
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

Allow creating DPP QP even if inline-data is not requested.  This is an
optimization to lower latency.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 1eebbb6ec3eb760f77240af7e850e93222e89afe)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
index ec310d2..6eedd2d 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_hw.c
@@ -2026,8 +2026,7 @@ int ocrdma_mbx_create_qp(struct ocrdma_qp *qp, struct ib_qp_init_attr *attrs,
     OCRDMA_CREATE_QP_REQ_RQ_CQID_MASK;
  qp->rq_cq = cq;
 
- if (pd->dpp_enabled && attrs->cap.max_inline_data && pd->num_dpp_qp &&
-     (attrs->cap.max_inline_data <= dev->attr.max_inline_data)) {
+ if (pd->dpp_enabled && pd->num_dpp_qp) {
   ocrdma_set_create_qp_dpp_cmd(cmd, pd, qp, enable_dpp_cq,
           dpp_cq_id);
  }
-- 
1.7.1